File: python-led/led.py
#!/usr/bin/python
import sys #reconocer lo que hay en la terminal
LED_PATH = "/sys/class/leds/beaglebone:green:usr3"

def writeLED(fileName, value, path=LED_PATH):
    """
    Ayuda de la rutina
    writeLED(fileName, value, path=LED_PATH)
    """
    fo = open(path + fileName, "w")
    fo.write(value)
    fo.close()
    return

def removeTrigger():
    writeLED("/trigger","none")
    return

print("Starting App")
#removeTrigger()
#writeLED("/brightness", "1")
#print("LED3 on")
#print("Finish App")
if len(sys.argv)!=2:
    print("Incorrect number of arguments")
    sys.exit(2)
    
if sys.argv[1] == "on":
    print("LED on")
    removeTrigger()
    writeLED("/brightness", "1")
    print("LED3 on")
elif sys.argv[1] == "off":
    print("LED off")
    removeTrigger()
    writeLED("/brightness", "0")
    print("LED3 off")
else:
    print("Wrong command!!!")
    
print("Script done")    

File: cpp-led/led.cpp
/* led.cpp */
#include"led.h"
#include<iostream>
#include<fstream>
#include<string>
#include<sstream>

#define LED_PATH "/sys/class/leds/beaglebone:green:usr"

CLED::CLED(int ledNumber){
	this->ledNumber = ledNumber; //Solo para argumentos que sean numeros
	std::ostringstream s;
	s << LED_PATH << ledNumber;
	path =std::string(s.str()); //convert to string
}

void CLED::WriteLED(std::string filename, std::string value){
	std::ofstream fs;
	fs.open((path+filename).c_str());
	fs << value;
	fs.close();
}

void CLED::RemoveTrigger(){
	WriteLED("/trigger", "none");
}

void CLED::TurnOn(){
	std::cout << "LED Turn On" << ledNumber << "on" << std::endl;
	RemoveTrigger();
	WriteLED("/brightness", "1");
}

CLED::~CLED(){
	std::cout << "destroying LED " << path << std::endl;
}
